File: src/main/java/com/osdu/model/delfi/geo/SpatialFilter.java
package com.osdu.model.delfi.geo;

import lombok.Data;

/**
 * GeoLocation object descriptor used by Delfi API.
 */
@Data
public class SpatialFilter {

    private static final String GEO_LOCATION_FIELD_ID = "data.dlLatLongWGS84";

    private String field = GEO_LOCATION_FIELD_ID;
    //Delfi spec assumes that there can be 1 of 3 different types of objects and in order to maintain that we need this property
    private GeoLocation byBoundingBox;
    private GeoLocation byDistance;
    private GeoLocation byGeoPolygon;

}

File: osdu-r2/os-python-sdk/osdu_api/model/search/spatial_filter.py
from osdu_api.model.search.by_bounding_box import ByBoundingBox
from osdu_api.model.search.by_distance import ByDistance
from osdu_api.model.search.by_geo_polygon import ByGeoPolygon

class SpatialFilter:
    def __init__(self, field: str, by_bounding_box: ByBoundingBox, by_distance: ByDistance, by_geo_polygon: ByGeoPolygon):
        self.field = field
        self.by_bounding_box = by_bounding_box
        self.by_distance = by_distance
        self.by_bounding_box = by_geo_polygon